stack/packaging-third-party-eups-dependencies.rst

```diff
@@ -22,10 +30,40 @@ See `this page`_ for a list of compatible licenses.
 .. _this page: https://www.gnu.org/licenses/license-list.html
 
 
+.. _fork-creating:
+
+Creating A Forked Repo
+======================
+
+A forked repo relies on building the package from source, rather than using binary artifacts, on the assumption that we may need to use the latest developments.
+Accordingly, we fork the package on GitHub into the ``lsst`` organization and add the "DM Externals" and "Overlords" teams to it as described in :doc:`adding-a-new-package`.
+This ensures that it will be tagged appropriately for official releases (while not conflicting with the third-party tags).
+
+Since the third-party package likely has no support for eups, we need to add a ``ups`` directory.
+This directory should contain the package's EUPS table file as well as an optional file :file:`eupspkg.cfg.sh` which will contain any customized commands for building and installing the third-party package.
+
+The upstream source has no need to know about the ``ups`` directory, so we put it on a separate branch in the ``lsst`` fork.
+That branch is traditionally named ``lsst-dev``.
+We set that branch to be the GitHub default in our fork.
+
+When changes are made to the upstream source, they can be merged into the ``lsst-dev`` branch, or ``lsst-dev`` can be rebased on an upstream branch.
+If we make local changes for our own needs, they can be converted into PRs to the upstream fork, leaving out any commits that refer to the ``ups`` directory contents.
+
+The new package then needs to be added to ``etc/repos.yaml`` in the ``lsst/repos`` repository.
+We indicate that the ``lsst-dev`` branch is the one that we will build from by adding a ``ref:`` clause in that file.
+
+Finally, the package should be added as an eups dependency to some other package or meta-package in its ``.table`` file.
```

Comment:
I'm still slightly confused about whether packages should be added as `setupOptional` or `setupRequired` in `lsst_distrib`. For example, `meas_extensions_multiprofit` will start off being used in an optional pipelineTask but (hopefully) become part of the DRP pipelines. At that point should it be required? I see that other measurement extensions are still optional, but multiprofit is like scarlet in that it's multiband and therefore not actually a measurement plugin. So really I'm wondering if meas_extensions_scarlet ought to be required now even if you can technically still turn it off entirely.

Reply:
For most purposes, there's effectively no difference between `setupOptional` and `setupRequired`.  They only differ in the case where someone has not installed an optional package in their local stack.

If you can do useful things with the depending package (and if its tests, if it has any, pass), then `setupOptional` can be a hint that it's not mandatory to have the dependency.  But the shared stack, CVMFS, stack containers, and the RSP will always have it anyway.